src/mlpack/methods/ann/layer/lambda_map_reduce_impl.hpp

```diff
@@ -0,0 +1,416 @@
+    reductionRule(std::move(other.reductionRule)),
+    layerOutputMatrix(std::move(other.layerOutputMatrix)),
+    layerDeltaMatrix(std::move(other.layerDeltaMatrix))
+{
+  // Ensure that the aliases for layers during passes have the right size.
+  layerOutputs.resize(network.size(), MatType());
+  layerDeltas.resize(network.size(), MatType());
+  layerGradients.resize(network.size(), MatType());
+
+  // layerOutputs, layerDeltas, and layerGradients will be reset the next time
+  // Forward(), Backward(), or Gradient() is called.
+
+  other.layerOutputs.clear();
+  other.layerDeltas.clear();
+  other.layerGradients.clear();
+}
+
+template<typename ReductionRuleType, typename MatType>
+LambdaMapReduceType<ReductionRuleType, MatType>& LambdaMapReduceType<ReductionRuleType, MatType>::operator=(const LambdaMapReduceType& other)
+{
+  if (this != &other)
+  {
+    Layer<MatType>::operator=(other);
+
+    network.clear();
+    layerOutputs.clear();
+    layerDeltas.clear();
+    layerGradients.clear();
+
+    inSize = other.inSize;
+    totalInputSize = other.totalInputSize;
+    totalOutputSize = other.totalOutputSize;
+    reductionRule = other.reductionRule;
+
+    layerOutputMatrix = other.layerOutputMatrix;
+    layerDeltaMatrix = other.layerDeltaMatrix;
+
+    for (size_t i = 0; i < other.network.size(); ++i)
+      network.push_back(other.network[i]->Clone());
+
+    // Ensure that the aliases for layers during passes have the right size.
+    layerOutputs.resize(network.size(), MatType());
+    layerDeltas.resize(network.size(), MatType());
+    layerGradients.resize(network.size(), MatType());
+  }
+
+  return *this;
+}
+
+template<typename ReductionRuleType, typename MatType>
+LambdaMapReduceType<ReductionRuleType, MatType>& LambdaMapReduceType<ReductionRuleType, MatType>::operator=(LambdaMapReduceType&& other)
+{
+  if (this != &other)
+  {
+    Layer<MatType>::operator=(other);
+
+    layerOutputs.clear();
+    layerDeltas.clear();
+    layerGradients.clear();
+
+    inSize = std::move(other.inSize);
+    totalInputSize = std::move(other.totalInputSize);
+    totalOutputSize = std::move(other.totalOutputSize);
+    reductionRule = std::move(other.reductionRule);
+
+    network = std::move(other.network);
+
+    layerOutputs.resize(network.size(), MatType());
+    layerDeltas.resize(network.size(), MatType());
+    layerGradients.resize(network.size(), MatType());
+
+    other.layerOutputs.clear();
+    other.layerDeltas.clear();
+    other.layerGradients.clear();
+  }
+
+  return *this;
+}
+
+template<typename ReductionRuleType, typename MatType>
+void LambdaMapReduceType<ReductionRuleType, MatType>::Forward(
+    const MatType& input, MatType& output)
+{
+  Forward(input, output, 0, network.size() - 1);
+}
+
+template<typename ReductionRuleType, typename MatType>
+void LambdaMapReduceType<ReductionRuleType, MatType>::Forward(
+    const MatType& input,
+    MatType& output,
+    const size_t start,
+    const size_t end)
```

Comment:
You could also avoid implementing this overload if you liked---I think it is only used by `FFN::Forward()`, which specifically calls it on its held `MultiLayer`.

Reply:
Ok.